bigquery_etl/shredder/delete.py

```diff
@@ -354,39 +345,46 @@ def main():
     client = client_q.default_client
     states = {}
     if args.state_table:
-        client.query(
-            dedent(
-                f"""
-                CREATE TABLE IF NOT EXISTS
-                  `{args.state_table}`(
-                    task_id STRING,
-                    job_id STRING,
-                    job_created TIMESTAMP,
-                    start_date DATE,
-                    end_date DATE
-                  )
-                """
-            ).strip()
-        )
-        states = {
-            row["task_id"]: row["job_id"]
-            for row in client.query(
-                dedent(
-                    f"""
-                    SELECT
-                      task_id,
-                      job_id,
-                    FROM
-                      `{args.state_table}`
-                    WHERE
-                      start_date = '{args.start_date}'
-                      AND end_date = '{args.end_date}'
-                    ORDER BY
-                      job_created
-                    """
-                ).strip()
-            ).result()
-        }
+        state_table_exists = False
+        try:
+            client.get_table(args.state_table)
+            state_table_exists = True
+        except NotFound:
+            if not args.dry_run:
+                client.create_table(
```

Comment:
this fixes dry run when using credentials that don't have permissions needed to create the state table